queue/queue_private.js

```diff
@@ -31,19 +46,35 @@ queueApp.controller("QueuePrivateController", function($scope, $controller, Queu
 		}
 
 		$scope.client.ended = true;
-		QueueService.finishAppointment(new Date().toISOString(), $scope.client.appointmentId, $scope.client.selectedStationNumber);
+		QueueService.finishAppointment(new Date().toISOString(), $scope.client.appointmentId, $scope.client.selectedStationNumber).then(function(result) {
+			if(!result.success) {
+				$scope.client.ended = false;
+				alert(result.error);
+			}
```

Comment:
If the appointment is successfully marked as completed, it might just sit in the list of appointments until that's updated, which could be a full minute. I think we should call ```$scope.updateAppointmentInformation();``` if the result is a success. And same thing after being marked as incomplete or cancelled.

Reply:
Hm I hate the idea of hitting the database and loading all the appts just to get the change that they literally just made... I will however make this more responsive, let me see what I can do.